### PR TITLE
Pr workflow updated for changed files

### LINKED ISSUE #1117: PR Workflow Enhancement - Delta Error/Warning Checks
Description
Currently, our PR Workflow checks the entire codebase for linting, formatting, and type errors, which can be time-consuming. This enhancement request is to optimize the workflow to only check for errors and warnings in the lines of code that are changed or added in the pull request.

Details:

Linting: Only check for linting errors and warnings in the modified or added lines of code.
Formatting: Run formatting checks only on the changes introduced in the pull request.
Type Errors: Optimize the type-checking process to focus on the modified or added code.
Context

This is the required check - PR Workflow / Check for linting, formatting and type errors (pull_request)
This enhancement aims to speed up the PR review process by narrowing down the scope of checks to the code that is directly affected by the pull request. It helps in quicker identification and resolution of issues related to the changes introduced in the PR.
Expected Behavior
The PR Workflow should only check for linting, formatting, and type errors in the modified or added lines of code.
The workflow should be optimized to reduce processing time for pull requests.

Additional Information

 - [x] I have reviewed the current PR Workflow configuration.
 - [x] I have considered potential impacts on existing workflows.
 - [x] This enhancement aligns with our project goals.

### PR DESCRIPTION
**Pull request by duplixx**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes  #1117

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Review by palisadoes — COMMENTED** [review]
1. Include the eslint configuration file with the desired changes.
1. How do you plan to get whether this works? I don't want to have the same problems repeated. As you and @tasneemkoushar are in the same timezone I'd suggest that you work together on a Zoom call to make sure subsequent PRs work as expected. For example using test PRs immediately after merging.

File: .github/workflows/pull-requests.yml [hunk 1]
```diff
@@ -44,34 +45,47 @@ jobs:
...
         run: npm run typecheck
-    
+
       - name: Run tests
         run: npm run test -- --watchAll=false --coverage
-  
+
+      - name: Get changed TypeScript files
+        id: changed-files
+        uses: actions/changed-files@v1
```
[the remark this answers is not in the document]
Comment (duplixx) [reply .github/workflows/pull-requests.yml:54]: yeah sure !! it will be great.

[the remark this answers is not in the document]
Comment (duplixx) [reply .github/workflows/pull-requests.yml]: fixed the changed file issue also added checked file condition

**Comment by duplixx** [comment]
@palisadoes 
yeah I have fixed the failing test,and the PR is ready to be reviewed 
Thanks!!

**Comment by palisadoes** [comment]
@anwersayeed please take a look

**Review by anwersayeed — APPROVED** [review]
Looks fine to me.

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
@duplixx

1. This needs to be merged against `develop` 
2. Please resubmit
3. Closing

**Pull request closed without merging**